### DISCUSSION TITLE
Accepted suggestions variables and functions are not auto importing.

### DISCUSSION THREAD

**Discussion by kanhaiya007** [community/community#6975] [Copilot Conversations]
Unlike regular IDE function which auto imports the suggestion when selected, there is a pain with Copilot while importing the suggestions.

**Comment by jansorg** [comment]
Which editor are you using?

**Reply by PalashTichkule** [reply]
I am using VS Code and seeing the same behavior.

**Reply by reader00** [reply]
I have the same problem in VS Code 1.61.2

**Reply by ctsstc** [reply]
Same for VS Code. Copilot's autocomplete does not seem to handle any auto imports that you would normally have occur if you used standard autocomplete. I'm sure it could know it needs to, but it simply auto completes the current line without realizing it needs to import given references.

Seems to be a duplicate here with more information/discussion: https://github.com/github/feedback/discussions/7388

**Reply by ggoonnzzaallo** [reply]
@reader00 I got frustrated enough and built an extension that fixes this. It:

1. Adds the import statement for the right library/function/variable when it's 100% clear
2. Automatically pops up IntelliSense when there's more than one possible solution

You can check it out [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot) if interested

**Comment by rnett** [comment]
I'm seeing this in IntelliJ as well.

**Comment by Thomas1664** [comment]
Same issue for me with vscode and React

**Comment by jeppard** [comment]
Same thing with vscode and typescript

**Comment by Josje92** [comment]
Almost two years down the road and still no way to fix this?

**Comment by ctsstc** [comment]
I've also noticed that Copilot now goes far enough out of its way to avoid Hallucinations to instead utilize imports and variables that already exist rather than trying to use something that would exist or match a current obvious pattern.

I understand it previously would try to do something like "import MagicFixesEverything" but now it has me working different to start with imports before letting Copilot try to write incorrect code that I have to change.

**Comment by ggoonnzzaallo** [comment]
It's been 1 year+ since I first encountered this bug and there's still no solution. 

I was so frustrated I actually built my own Copilot equivalent that handles imports correctly by:
1. Adding the import statement for the right library/function/variable when it's 100% clear
2. Automatically pop-up IntelliSense when there's more than one possible solution

If anyone is interesting to try it, you can download the extension [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot).

**Reply by tendiesfortwo** [reply]
Looks good, will try it
